### ISSUE TITLE
Accounts can be created with emails that are already verified in the system

### ISSUE THREAD

**Issue by ElizabethStirling** [sourcegraph/sourcegraph#22447]
- **Sourcegraph version:** 
Sourcegraph.com
- **Platform information:** 
Firefox version 89.02 on macOS Big Sur 11.4

Tl;dr - Using builtin authentication, users can create accounts with emails that are already verified. This then creates an account in an unhealthy state that can only be signed in to via username. The new account's email cannot be verified, and attempting to do so causes an error. They should instead be prompted to use their existing account.

#### Steps to reproduce:

1. Create an account on sourcegraph.com through our builtin authentication (account A) with a testing email (ex. foo@bar.com)
2. Verify account A's email. Now account A has a verified primary email (foo@bar.com)
3. Create a new account via builtin authentication (account B) with the same email (foo@bar.com), and a different password & username
4. Try to verify account B's email - this will throw an error
5. Try to sign in to account B via email - this should fail. If it does not, then I would expect account A to be locked out
6. Try to sign in to account B via username - this should succeed
7. Try to sign in to account A via email - this should work. If it does not, then I would expect step 5 to have succeeded.

#### Expected behavior:
The user will be informed that an account with that email already exists. Prompt them to either sign in with their password, reset their password, or sign in with the relevant codehost account. The user should be informed that an account exists even if they haven't verified their email on any account, and should be able to reset their password.

#### Actual behavior:
Account B is created in an unhealthy state where its email can't be verified. Additionally, the user cannot sign in to account B by using their email - they must use the username they provided during the creation process.

**Comment by artemruts** [comment]
@artemruts TODO: import to Jira

**Comment by artemruts** [comment]
@ElizabethStirling in the current sprint now: https://sourcegraph.atlassian.net/browse/COREAPP-133
